### DISCUSSION TITLE
Version 2.x

### DISCUSSION THREAD

**Discussion by jhollowe** [proxmoxer/proxmoxer#93] [Announcements]
I'd like to announce plans to move to a new major version of proxmoxer: version 2.x

## Planned Breaking Changes
There are some things that have needed to be changed or updated that have not been due to it causing breaking changes. This new major version will allow the (small and few) breaking changes to take place. These changes will be well documented with a migration guide added to the documentation.

This will also be the version that will remove support for older versions of Python. As of right now, 3.7 is the oldest version supported by the Python Software Foundation and will thus be the oldest version of Python targeted for proxmoxer. Python 2.7 support will be removed to allow for simpler code, dependencies, and usage of newer language features.

## 1.x Support 
The existing 1.x codeline will only be supported for critical security bugs should they arrise. New features and bugs will be only fixed in future 2.x versions. 

## New Stuff
With version 2 will come more complete documentation with examples and common usecases. Also, a new `proxmoxer.tools` group of functionality will be added to simplify and standardize common tasks for which proxmoxer is used. The test system will also be updated to a modern framework to allow cleaner tests and more rebust and in-depth testing.

## Your Part
The documentation and tools functionality can only be made useful if they reflect what the community is doing.
Please be involved: requests features, submit code, add documentation for the things you use proxmoxer for, provide feedback on what works and what needs improvement.

**Comment by gaby** [comment]
@jhollowe Any plans to do a new release? It's been over a year since 2.x came out

**Reply by jhollowe** [reply]
Are there features/changes since 2.0.1 that you are waiting for?

**Reply by gaby** [reply]
@jhollowe Mostly py3.12 support, although I think the current latest package works with 3.12? Thanks!
